web/src/cypress/support/schedule.ts

```diff
@@ -138,6 +145,79 @@ function deleteSchedule(id: string): Cypress.Chainable<void> {
   })
 }
 
+function createTemporarySchedule(
+  scheduleID?: string,
+  options?: TemporaryScheduleOptions,
+): Cypress.Chainable<null> {
+  const mutation = `
+    mutation($input: SetTemporaryScheduleInput!) {
+      setTemporarySchedule(input: $input)
+    }
+  `
+
+  if (!scheduleID) {
+    return cy
+      .createSchedule()
+      .then((s: Schedule) => createTemporarySchedule(s.id, options))
+  }
+
+  let input = omit(options ?? {}, 'shiftUserIDs')
+  input.scheduleID = scheduleID
+
+  const now = DateTime.local()
+  const r = (min: number, max: number): number => c.integer({ min, max })
+
+  const MAX_FUTURE = 1576800 // up to 3 years (in minutes) in the future
```

Comment:
This is an arbitrary date, since we probably don't need to functionally validate a temporary schedule a user makes more than 3 years out (since likely all temporary schedules made will take place within a few months)